astrocut/make_cube.py

```diff
@@ -346,3 +349,500 @@ def make_cube(self, file_list, cube_file="img-cube.fits", sector=None, max_memor
+        
+
+        self.cube_file = cube_file
+
+    def _write_block(self, cube_hdu, start_row=0, end_row=None, fill_info_table=False, verbose=False):
+        """ Do one pass through the image files and write a block of the cube.
+
+        cube_hdu is an hdulist object opened in update mode
+        """
+
+        # Initializing the sub-cube
+        nrows = (self.cube_shape[0] - start_row) if (end_row is None) else (end_row - start_row)
+        sub_cube = np.zeros((nrows, *self.cube_shape[1:]), dtype=np.float32)
+        
+        # Loop through files
+        for i, fle in enumerate(self.file_list):
+
+            if verbose:
+                st = time()
+
+            with fits.open(fle, mode='denywrite', memmap=True) as ffi_data:
+
+                # add the image and info to the arrays
+                sub_cube[:, :, i, 0] = ffi_data[0].data[start_row:end_row, :]
+
+                del ffi_data[0].data
+               
+                if fill_info_table:  # Also save the header info in the info table
+
+                    for kwd in self.info_table.columns:  # Iterate over every keyword in the TICA FFI primary header
+                        if kwd == "FFI_FILE":
+                            self.info_table[kwd][i] = os.path.basename(fle)
+                            
+                        else:
+                            nulval = None
+                            if self.info_table[kwd].dtype.name == "int32":
+                                nulval = 0
+                            elif self.info_table[kwd].dtype.char == "S":  # hacky way to check if it's a string
+                                nulval = ""
+                            
+                            # NOTE: 
+                            # A header keyword ('COMMENT') in TICA returns a keyword 
+                            # value in the form of a _HeaderCommentaryCard instead of a STRING. 
+                            # This breaks the info table because the info table 
+                            # doesn't understand what a FITS header commentary card is. 
+                            # Adding a try/except is a way to catch when these instances happen 
+                            # and turn the keyword value from a _HeaderCommentaryCard to a
+                            # string, which is what it's meant to be in the info table. 
+                            #
+                            # TO-DO: Find a more elegant way to handle these stray 
+                            # _HeaderCommentaryCards. 
+                            
+                            try:
+                                self.info_table[kwd][i] = ffi_data[0].header.get(kwd, nulval)
+                            except ValueError:
+                                kwd_val = ffi_data[0].header.get(kwd)
+                                if isinstance(kwd_val, fits.header._HeaderCommentaryCards):
+                                    self.info_table[kwd][i] = str(kwd_val)
```

Comment:
if we raise a valueerror here and it's not an instance of _headerCOmmentaryCards, we just silently ignore the error.

We should add a `raise` here at the end

```suggestion
                            except ValueError:
                                kwd_val = ffi_data[0].header.get(kwd)
                                if isinstance(kwd_val, fits.header._HeaderCommentaryCards):
                                    self.info_table[kwd][i] = str(kwd_val)
                                else:
                                    raise
```